src/LanguageServer/Impl/Indexing/IndexParser.cs

```diff
@@ -37,34 +41,64 @@ internal sealed class IndexParser : IIndexParser {
             _version = version;
         }
 
-        public void Dispose() {
-            _allProcessingCts.Cancel();
-            _allProcessingCts.Dispose();
+        public Task ParseAsync(string path, CancellationToken cancellationToken = default) {
+            lock (_syncObj) {
+                CancelCurrentParse();
```

Comment:
You are abandoning previous parse which is OK. However, the task reads the file. What happens if you start new task that also wants to read same file. You may want to allow shared reads by specifying explicit modes: `FileMode.Open, FileAccess.Read, FileShare.Read`